### PR TITLE
Matter Water Sensor Device Integration

### PR DESCRIPTION
**Pull request by hcarter-775**
This PR includes protoype support for:
* Water Leak Detector
* Water Freeze Detector
* Rain Sensor

### CODE REVIEW

File: drivers/SmartThings/matter-thermostat/profiles/air-purifier-hepa-ac-wind-thermostat-humidity-fan-heating-only-nostate-nobattery-aqs-pm10-pm25-ch2o-meas-pm10-pm25-ch2o-no2-tvoc-level.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
+name: air-purifier-hepa-ac-wind-thermostat-humidity-fan-heating-only-nostate-nobattery-aqs-pm10-pm25-ch2o-meas-pm10-pm25-ch2o-no2-tvoc-level
```
Comment (HunsupJung) [inline drivers/SmartThings/matter-thermostat/profiles/air-purifier-hepa-ac-wind-thermostat-humidity-fan-heating-only-nostate-nobattery-aqs-pm10-pm25-ch2o-meas-pm10-pm25-ch2o-no2-tvoc-level.yml:1]: Wow, It's really long name. is it okay? actually I don't know about name policy well

File: drivers/SmartThings/matter-water-sensor/config.yml
Comment (HunsupJung) [inline drivers/SmartThings/matter-water-sensor/config.yml:1]: Can I know why is it empty?

**Comment by ctowns** [comment]
Hey @hcarter-775 could remove the non-Water Sensor files from this PR for easier review? I think the only commit that is needed is [9644ec8](https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1518/commits/9644ec89c04a8eb98677b8b8dcca0b50615ca152)

The rest are old AQS matter-thermostat commits that do not need to be reviewed here since they have already been squashed and commited to main.

**Comment by hcarter-775** [comment]
Closed: This is an older version of the current 1.3 water sensor changes. The corrected water sensor changes are posted in [PR: 1533](https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1533).

**Pull request closed without merging**